### PR TITLE
feat: loader component bb-19

### LINKED ISSUE #19: feat: Loader component
- [x] Create a common LoaderWrapper component - it will be used while the data is fetching.

### PR DESCRIPTION
**Pull request by ToyLess78**
PR Title:
feat: Create LoaderWrapper component

PR Description:

This PR introduces a new LoaderWrapper component that will be used as a common loading indicator across the application. The LoaderWrapper component is designed to provide a consistent and reusable way to display a loading state while data is being fetched.

Changes include:

Created LoaderWrapper component that utilizes 
Integrated the LoaderWrapper component within the Auth screen to display the loader when dataStatus is "pending".

Issue Reference:
Resolves - Create a common LoaderWrapper component to be used while data is fetching.

Additional Notes:

This component can be easily extended or customized in the future if additional functionality is required.
Please review the changes and provide feedback if necessary.
Знімок екрана 2024-08-19 о 22 37 45

### CODE REVIEW

**Comment by anogire** [comment]
Please, delete issue from your PR. It shouldn't automatically close.

File: apps/mobile/src/libs/components/common/loader-wrapper/loader-wrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+		}
+		return children;
+	};
+
+	return (
+		<View style={[globalStyles.flex1, globalStyles.justifyContentCenter]}>
+			{renderContent()}
+		</View>
+	);
+};
```
Comment (revenets) [inline apps/mobile/src/libs/components/common/loader-wrapper/loader-wrapper.tsx:26]: You may not want to affect the layout of the child component. And I believe we can get rid of the additional wrapper for ActivityIndicator.
```suggestion
	return isLoading ? (
		  <ActivityIndicator 
		      color={BaseColor.BG_BLUE}
		      size="large"
		      style={[
		          globalStyles.flex1,
		          globalStyles.justifyContentCenter,
		          globalStyles.alignItemsCenter
		      ]} />
	) : (
		<>{children}</>
	);
};
```

**Review by revenets — CHANGES_REQUESTED** [review]

**Comment by ToyLess78** [comment]
Знімок екрана 2024-08-19 о 22 37 45

File: apps/mobile/src/libs/components/common/loader-wrapper/loader-wrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import { ActivityIndicator } from "react-native";
+
+import { BaseColor } from "~/libs/enums/app/base-color.enum";
+import { globalStyles } from "~/libs/styles/styles";
```
Comment (anogire) [inline apps/mobile/src/libs/components/common/loader-wrapper/loader-wrapper.tsx:5]: import { BaseColor } from "~/libs/enums/enums"

File: apps/mobile/src/screens/auth/auth.tsx [hunk 3]
```diff
@@ -55,12 +59,12 @@ const Auth: React.FC = () => {
 	};
 
 	return (
-		<>
+		<LoaderWrapper isLoading={dataStatus === "pending"}>
 			<ScreenWrapper>
```
Comment (anogire) [inline apps/mobile/src/screens/auth/auth.tsx:63]: Let's use DataStatus from "~/libs/enums/enums"

File: apps/mobile/src/libs/components/common/loader-wrapper/loader-wrapper.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import { ActivityIndicator } from "react-native";
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/common/loader-wrapper/loader-wrapper.tsx:2]: export from ~/libs/components/components

File: apps/mobile/src/libs/components/components.tsx [hunk 5]
```diff
@@ -1,5 +1,6 @@
 export { App } from "./app/app";
 export { Button } from "./button/button";
+export { LoaderWrapper } from "./common/loader-wrapper/loader-wrapper";
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/components.tsx:3]: The components in this file are already common, no need to create `common` folder

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by revenets — APPROVED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -2,16 +2,15 @@
...
-* colors enum bb-23 ([#36](https://github.com/BinaryStudioAcademy/bsa-2024-bebalance/issues/36)) ([a5c2441](https://github.com/BinaryStudioAcademy/bsa-2024-bebalance/commit/a5c24416e6e46d09dd02c1d813db0b99df7fa423))
-* Custom fonts and Text component bb-18 ([#38](https://github.com/BinaryStudioAcademy/bsa-2024-bebalance/issues/38)) ([959c621](https://github.com/BinaryStudioAcademy/bsa-2024-bebalance/commit/959c6218913aefdf6cc35ad899fab69f80eb47e2))
-* global styles bb-17 ([#35](https://github.com/BinaryStudioAcademy/bsa-2024-bebalance/issues/35)) ([d712a34](https://github.com/BinaryStudioAcademy/bsa-2024-bebalance/commit/d712a34336cc403d455d194cc69d719f55025c1e))
-* protected routing bb-12 ([#39](https://github.com/BinaryStudioAcademy/bsa-2024-bebalance/issues/39)) ([641f701](https://github.com/BinaryStudioAcademy/bsa-2024-bebalance/commit/641f70179bb200f12c6a5833557a0cb494a39979))
+- add continuous delivery bb-3 ([#62](https://github.com/BinaryStudioAcademy/bsa-2024-bebalance/issues/62)) ([4106756](https://github.com/BinaryStudioAcademy/bsa-2024-bebalance/commit/4106756bc7ecd9119ecf302b5d0df67089aa5961))
```
Comment (fshabanov) [inline CHANGELOG.md:7]: Undo these changes, add .prettierignore as it was written in the internal chat

File: apps/mobile/src/libs/components/loader-wrapper/loader-wrapper.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+
+import { ActivityIndicator } from "~/libs/components/components";
+import { BaseColor } from "~/libs/enums/app/base-color.enum";
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/loader-wrapper/loader-wrapper.tsx:4]:
```suggestion
import { BaseColor } from "~/libs/enums/enums";
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by fshabanov — APPROVED** [review]

**Pull request merged**